### ISSUE TITLE
Regression deserializing string Enums between 1.0.3 and 1.0.4

### ISSUE THREAD

**Issue by jgraham** [serde-rs/serde#932]
The example in https://gist.github.com/anonymous/440028c896809d8b3b51d8333cb6d2f9 works for me with `serde = "=1.0.3"`, `serde_json = "1.0.2"` and `serde_derive = "1.0.4"` and current stable rust, but errors when specifying `serde = "=1.0.4"` (or less precisely, just setting all those things to `= 1`.

The output I see is:
```
{"action":"log","level":"INFO"}
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ErrorImpl { code: Message("invalid type: string \"INFO\", expected string or map"), line: 0, column: 0 }', /checkout/src/libcore/result.rs:859
```

**Comment by jgraham** [comment]
Thanks!

**Comment by dtolnay** [comment]
I published serde 1.0.5 with the fix.
